src/subclass/widget.rs

```diff
@@ -209,6 +210,10 @@ pub trait WidgetImpl: WidgetImplExt + ObjectImpl + 'static {
     fn get_preferred_height_for_width(&self, widget: &Widget, width: i32) -> (i32, i32) {
         self.parent_get_preferred_height_for_width(widget, width)
     }
+
+    fn size_allocate(&self, widget: &Widget, allocation: &Allocation) {
```

Comment:
The function `gtk_widget_size_allocate()` says in the docs:

> In this function, the allocation may be adjusted. It will be forced to a 1x1 minimum size, and the adjust_size_allocation virtual method on the child will be used to adjust the allocation. Standard adjustments include removing the widget’s margins, and applying the widget’s Widget:halign and Widget:valign properties.

So maybe this needs to be mutable after all and subclasses need to be able to modify it?

Reply:
Note that I have no idea how this function is supposed to work, I'm just guessing from the docs :)

Reply:
That was my initial guess (see previous PR).

Looking at https://gitlab.gnome.org/GNOME/gtk/blob/gtk-3-24/gtk/gtkwidget.c#L6443
The GtkWidget vfunc doesn't modify it.

Reply:
Does any of the subclasses? Maybe worth asking on #gtk :)

Reply:
From what I can see they don't. At best they call gtk_widget_set_allocation or the parent class size allocate vfunc.

Reply:
Would be useful to get a definite answer from a GTK developer :)

Reply:
@ebassi maybe? :)